frontend/src/router/index.ts

```diff
@@ -21,42 +22,63 @@ import SummaryView from '@/views/SummaryView.vue';
 // 3. (Not recommended) Within router below, use route-level code-splitting which generates a separately loaded javascript file for this route. Syntax: component: () => import(../components/<component>.vue) syntax.
 //    This fixes the issue, but seems to break using shared state (e.g. in ApplicationService).
 
+const breadcrumbRouteItems = {
+    dashboard: {
+        to: '/dashboard',
+        label: 'Manage Permissions',
+    },
+    addUserPermission: {
+        to: '/grant',
+        label: 'Add User Permission'
+    }
+} as Breadcrumb;
+
```

Comment:
The "[xxx]Items" is really attractive to me, I like it.
I am thinking, maybe you can even make them better by instead of giving it breadcrumb items, can we give it as some **basic "route items"** (since breadcrumb depends on 'route')? Rname and reusing the Interface and type you defined in the state (and probably move them here).
Let me know and we can discuss if this is approachable.

Reply:
seens like a great idea, will do!

Reply:
Code change looks good. Thanks Jota!